### PR TITLE
User creation

### PR DESCRIPTION
**Pull request by yuvala15**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "test": "cross-env NODE_ENV=test mocha --recursive -r ts-node/register src/**/*.spec.ts --exit",
     "test:validation": "cross-env NODE_ENV=test mocha --recursive -r ts-node/register \"src/**/*.validator.spec.ts\" --exit",
     "test:integration": "cross-env NODE_ENV=test mocha --recursive -r ts-node/register \"src/**/*.router.spec.ts\" --exit",
-    "test:repository": "cross-env NODE_ENV=test mocha --recursive -r ts-node/register \"src/**/*.repository.spec.ts\" --exit",
+    "test:repository": "cross-env NODE_ENV=test mocha --recursive --timeout 5000 -r ts-node/register \"src/**/*.repository.spec.ts\" --exit",
```
Comment (Ron537) [inline package.json]: No external services are used here. No need to change default timeout. If you get timeout errors, there is something wrong with the implementation

File: src/config.ts [hunk 2]
```diff
@@ -20,10 +20,32 @@ export const config = {
         password: process.env.RMQ_PASSWORD || 'guest',
         username: process.env.RMQ_USERNAME || 'guest',
     },
+    validator: {
+        user: {
+            id: /^[\w-]+@[\w-]+$/,
```
Comment (Ron537) [inline src/config.ts]: In my opinion the standard for user unique id is `^\w+@\w+$\` no need for `-`.
Prove me wrong and I'll accept this.

File: src/config.ts [hunk 3]
```diff
@@ -20,10 +20,32 @@ export const config = {
...
+            email: /^[\w-]+\@[\w-]+\.[\w-]+$/i,
+        },
+    },
     server: {
-        port:  +(process.env.PORT || 3000),
+        port: +(process.env.PORT || 3000),
         name: 'user',
     },
+    router: {
+        getMany: {
```
Comment (Ron537) [inline src/config.ts]: Here you can maybe separate it to 2 different objects:
``` javascript
pagination: {
     startIndex: number,
     endIndex: number,
},
sort: {
    sortOrder: '' | '-',
    sortBy: string
}
```

File: src/user/user.interface.ts [hunk 4]
```diff
@@ -1,4 +1,6 @@
 export interface IUser {
-    id?: string;
-    property: string;
+    _id: string;
```
Comment (Ron537) [inline src/user/user.interface.ts]: Change to `id` don't use `_id`.

File: src/user/user.manager.ts [hunk 5]
```diff
@@ -26,12 +18,8 @@ export class UserManager {
         return UserRepository.getById(id);
     }
 
-    static getOne(userFilter: Partial<IUser>) {
-        return UserRepository.getOne(userFilter);
-    }
-
-    static getMany(userFilter: Partial<IUser>) {
-        return UserRepository.getMany(userFilter);
+    static getMany(userFilter: Partial<IUser>, startIndex: number, endIndex: number, sortOrder: string, sortBy: string) {
```
Comment (Ron537) [inline src/user/user.manager.ts]: `startIndex`, `endIndex`, `sortOrder` and `sortBy`should be optional fields in this layer as the controller might not pass all those properties and the repository as default value for undefined values

File: src/config.ts (hunk 2 above)
Comment (yuvala15) [reply src/config.ts]: I proved you!!

File: package.json (hunk 1 above)
Comment (yuvala15) [reply package.json]: Done.

File: src/user/user.manager.ts (hunk 5 above)
Comment (yuvala15) [reply src/user/user.manager.ts]: Fixed.

File: src/config.ts (hunk 3 above)
Comment (yuvala15) [reply src/config.ts]: Ok, I changed it.

**Pull request merged**